includes/admin/class-wc-rest-stripe-account-keys-controller.php

```diff
@@ -399,8 +399,10 @@ public function configure_webhooks( WP_REST_Request $request ) {
 			$secret = $saved_secret;
 		}
 
+		WC_Stripe_API::set_secret_key( $secret );
+
 		try {
-			$response = $this->account->configure_webhooks( $environment, $secret );
+			$response = $this->account->configure_webhooks( $environment );
 		} catch ( Exception $e ) {
 			return new WP_REST_Response( [ 'message' => $e->getMessage() ], 400 );
 		}
```

Comment:
Should we add a `finally` clause that ensures we reset the secret key?
```suggestion
		} finally {
			// Ensure we reset the key before we do anything else.
			WC_Stripe_API::set_secret_key( '' );
		}
```

Reply:
Make sense, as it would avoid having stale keys in options. 

Addressed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4747/commits/00a70db15482befc9223e1362e6a58127e2b2cdf (invoked when clicking the reconfigure webhook button) and https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4747/commits/c9b0f384e3c8984a0c891880ca292c9cd54266f3 (invoked during onboarding flow)